src/main/java/edu/tseidler/model/Board.java

```diff
@@ -4,75 +4,73 @@
 
 public class Board {
     private static final int MINIMUM_SIZE_AND_WINNING_NUMBER = 3;
-    private final int maxRow;
-    private final int maxCol;
-    private final int winningNumber;
+    private final BoardParameters boardParameters;
     private final Fields fields;
     final Drawer drawer;
 
-    public Board(int[] dimensionsAndWinningNumber) {
-        maxRow = Math.max(MINIMUM_SIZE_AND_WINNING_NUMBER, dimensionsAndWinningNumber[0]);
-        maxCol = Math.max(MINIMUM_SIZE_AND_WINNING_NUMBER, dimensionsAndWinningNumber[1]);
-        this.winningNumber = Math.max(
+    public Board(int maxRow, int maxCol, int winningNumber) {
+        int maxRowValidated = Math.max(MINIMUM_SIZE_AND_WINNING_NUMBER, maxRow);
+        int maxColValidated = Math.max(MINIMUM_SIZE_AND_WINNING_NUMBER, maxCol);
+        int winningNumberValidated = Math.max(
                 MINIMUM_SIZE_AND_WINNING_NUMBER,
-                getPossibleWinningNumber(dimensionsAndWinningNumber[2]));
+                getPossibleWinningNumber(maxRowValidated, maxColValidated, winningNumber));
+        boardParameters = new BoardParameters(maxRowValidated, maxColValidated, winningNumberValidated);
         fields = new Fields();
```

Comment:
There is a saying that dependency injection is great

Example explanation: https://typeofweb.com/2016/07/07/wzorce-projektowe-dependency-injection/

Reply:
Dzięki za wskazanie błędu, mam nadzieję, że nowe zmiany załatwiają ten problem :)